modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java

```diff
@@ -709,4 +657,43 @@ public List<ServiceVo> listDeployedServicesOfIsv(ServiceQueryModel query) {
                         .equals(deployServiceEntity.getNamespace()))
                 .map(this::convertToServiceVo).toList();
     }
+
+    /**
+     * Get deploy service detail by id.
+     *
+     * @param id ID of deploy service.
+     * @return serviceDetailVo
+     */
+    public ServiceDetailVo getServiceDetailsByIdForIsv(UUID id) {
+        DeployServiceEntity deployServiceEntity = getDeployServiceEntity(id);
+        ServiceHostingType serviceHostingType =
+                deployServiceEntity.getDeployRequest().getServiceHostingType();
+        if (ServiceHostingType.SERVICE_VENDOR == serviceHostingType) {
+            String errorMsg = String.format("ServiceHostingType: %s of the Service with id %s not "
+                    + "found.", serviceHostingType, id);
+            log.error(errorMsg);
+            throw new ServiceHostingTypeNotFoundException(errorMsg);
```

Comment:
if we want to define a new exception in this case, then ServiceDetailsNotAccessible 403 is a better name.